### PR TITLE
portalicious: status change popup

### PR DESCRIPTION
**Pull request by PeterSmallenbroek**
[AB#30837](https://dev.azure.com/redcrossnl/7d17f2a9-3f18-40f1-8c29-98568fbca7e4/_workitems/edit/30837) 

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have added tests wherever relevant
- [ ] I have made sure that all automated checks pass before requesting a review
- [ ] I do not need any deviation from our PR guidelines

### CODE REVIEW

File: interfaces/Portalicious/src/locale/messages.xlf [hunk 1]
```diff
@@ -920,6 +920,123 @@
...
+        <source>The action &quot;Decline&quot; can not be applied to registrations with the &quot;Completed&quot; status. Registrations with “completed” status will remain with their current status.</source>
+      </trans-unit>
+      <trans-unit id="6666596645491746994" datatype="html">
+        <source>This means that they will not included in payments.</source>
+      </trans-unit>
+      <trans-unit id="6759205696902713848" datatype="html">
+        <source>Warning</source>
+      </trans-unit>
+      <trans-unit id="680137431344473981" datatype="html">
+        <source>areYouSure</source>
```
Comment (aberonni) [inline interfaces/Portalicious/src/locale/messages.xlf:1011]: This seems like an error?

File: interfaces/Portalicious/src/app/pages/project-registrations/components/change-status-dialog/change-status-dialog.component.html [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+        class="pt-2"
+      >
+        Select registrations with another status and try again.
+      </p>
+    }
+  }
+  <div class="mt-6 flex justify-end space-x-2">
+    <p-button
+      label="OK"
+      i18n-label="@@generic-ok"
```
Comment (aberonni) [inline interfaces/Portalicious/src/app/pages/project-registrations/components/change-status-dialog/change-status-dialog.component.html:298]:
```suggestion
      label="Cancel"
      i18n-label="@@generic-cancel"
```

Better IMHO but I don't feel strongly about it

File: interfaces/Portalicious/src/app/domains/registration/registration.api.service.ts [hunk 3]
```diff
@@ -85,6 +88,61 @@ export class RegistrationApiService extends DomainApiService {
...
+        ...body,
+        message: messageData.customMessage,
+        status,
+      };
+    } else if (messageData) {
+      body = {
+        ...body,
+        messageTemplateKey: messageData.messageTemplateKey,
+      };
+    }
```
Comment (aberonni) [inline interfaces/Portalicious/src/app/domains/registration/registration.api.service.ts:119]:
```suggestion
    if (messageData && 'customMessage' in messageData) {
      body.message = messageData.customMessage;
    } else if (messageData) {
      body.messageTemplateKey = messageData.messageTemplateKey;
    }
```

Maybe slightly easier to read?

File: interfaces/Portalicious/src/app/domains/registration/registration.api.service.ts [hunk 4]
```diff
@@ -85,6 +88,61 @@ export class RegistrationApiService extends DomainApiService {
...
+    projectId: Signal<number>;
+    paginateQuery: PaginateQuery | undefined;
+    status: RegistrationStatusEnum;
+    messageData?: SendMessageData | undefined;
+    dryRun: boolean;
+  }) {
+    let body: RegistrationStatusPatchDto;
+    body = {
+      status,
+    };
```
Comment (aberonni) [inline interfaces/Portalicious/src/app/domains/registration/registration.api.service.ts:107]:
```suggestion
    let body: RegistrationStatusPatchDto = {
      status,
    };
```

Bit of a nitpick but not sure why it needs a separate line.

File: interfaces/Portalicious/src/app/pages/project-registrations/components/change-status-contents-with-custom-message/change-status-contents-with-custom-message.component.html [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    />
+    <p-button
+      label="Approve"
+      i18n-label="@@generic-approve"
+      rounded
+      [disabled]="isMutating()"
+      [loading]="isMutating()"
+      type="submit"
+    />
+  </div>
```
[the remark this answers is not in the document]
Comment (aberonni) [reply interfaces/Portalicious/src/app/pages/project-registrations/components/change-status-contents-with-custom-message/change-status-contents-with-custom-message.component.html:62]: @PeterSmallenbroek looks like you forgot to reference the new component here.

**Review by aberonni — CHANGES_REQUESTED** [review]
Some more minor things.

File: interfaces/Portalicious/src/app/pages/project-registrations/components/change-status-contents-with-custom-message/change-status-contents-with-custom-message.component.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      return;
+    },
+  });
+
+  ngOnInit(): void {
+    this.previewData.set(undefined);
+  }
+
+  cancelClick() {
+    this.previewData.set(undefined);
```
Comment (PeterSmallenbroek) [inline interfaces/Portalicious/src/app/pages/project-registrations/components/change-status-contents-with-custom-message/change-status-contents-with-custom-message.component.ts:76]:
```suggestion
```

File: interfaces/Portalicious/src/app/pages/project-registrations/components/change-status-dialog/change-status-dialog.component.ts [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+    if (!status) {
+      return '';
+    }
+    return REGISTRATION_STATUS_VERB[status];
+  });
+  canSendMessage = computed(() => {
+    const status = this.status();
+    if (!status) {
+      return false;
+    }
```
Comment (PeterSmallenbroek) [inline interfaces/Portalicious/src/app/pages/project-registrations/components/change-status-dialog/change-status-dialog.component.ts:113]:
```suggestion
```

File: interfaces/Portalicious/src/app/pages/project-registrations/components/change-status-dialog/change-status-dialog.component.ts [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+  }
+
+  triggerAction(
+    actionData: ActionDataWithPaginateQuery<Registration>,
+    status: RegistrationStatusEnum,
+  ) {
+    this.actionData.set(actionData);
+    // Doing this to trigger the effect
+    this.status.set(undefined);
+    this.status.set(status);
```
[the remark this answers is not in the document]
Comment (PeterSmallenbroek) [reply interfaces/Portalicious/src/app/pages/project-registrations/components/change-status-dialog/change-status-dialog.component.ts:154]: See AB#31286

File: interfaces/Portalicious/src/app/pages/project-registrations/components/change-status-dialog/change-status-dialog.component.html [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+          }
+        }
+      }
+    </div>
+  </form>
+</p-dialog>
+<app-confirmation-dialog
+  #dryRunWarningDialog
+  header="Warning"
+  i18n-header
```
Comment (aberonni) [inline interfaces/Portalicious/src/app/pages/project-registrations/components/change-status-dialog/change-status-dialog.component.html:111]:
```suggestion
  i18n-header="@@generic-warning"
```

**Review by aberonni — APPROVED** [review]

**Review by aberonni — APPROVED** [review]

**Pull request merged**